lib/compiler/index.js

```diff
@@ -48,6 +50,10 @@ const compiler = {
   // or modify it as needed. If the pass encounters a semantic error, it throws
   // |peg.GrammarError|.
   passes: {
+    templates: [
+      instantiateTemplates,
+      removeNonInstantiatedTemplates,
+    ],
```

Comment:
It would be better to work with templates after doing some basic checks, like undefined or duplicating rules